### PR TITLE
feature: add an example plugin cli

### PR DESCRIPTION
**Pull request by jimlambrt**
Just something simple to demo how build an app that uses go-plugin
wrappers.

### CODE REVIEW

File: examples/plugin-cli/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# plugin-cli
+An example CLI that demonstrates how to incorporate a
+[go-plugin](https://github.com/hashicorp/go-plugin) wrapper into an application.
+In this example, we'll use a vault transit go-plugin.
```
Comment (jefferai) [inline examples/plugin-cli/README.md:4]:
```suggestion
In this example, we'll use a Vault `transit` plugin.
```

File: examples/plugin-cli/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Why would you want to use a vault transit via a
```
Comment (jefferai) [inline examples/plugin-cli/README.md:6]: Same -- we should capitalize Vault

File: examples/plugin-cli/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+[go-plugin](https://github.com/hashicorp/go-plugin) vs just including the
```
Comment (jefferai) [inline examples/plugin-cli/README.md:7]:
```suggestion
[go-plugin](https://github.com/hashicorp/go-plugin) vs. just including the
```

File: examples/plugin-cli/README.md (hunk 2 above)
Comment (jefferai) [reply examples/plugin-cli/README.md:6]:
```suggestion
Why would you want to use Vault `transit` via a
```

File: examples/plugin-cli/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+to allow users to chose from a variety of KMS wrappers within your application,
+via configuration but you didn't want to include every possible KMS wrapper
+dependency into your application.  Using go-plugin wrappers allows you to build
+a set of kms wrappers as plugins and perhaps embed the plugin binaries into your
+app (like we do in this example).  Then your app has no direct dependencies on
+the KMS's you wish to support.
+
+
+Running the cli will:
+- Initialize a go-plugin vault transit.
```
Comment (jefferai) [inline examples/plugin-cli/README.md:18]:
```suggestion
- Initialize a Vault `transit` plugin.
```

File: examples/plugin-cli/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+- Encrypt a plaintext secret using the plugin vault transit wrapper.
+- Decrypt the cipher text secret using the plugin vault transit wrapper.
```
Comment (jefferai) [inline examples/plugin-cli/README.md:20]:
```suggestion
- Decrypt the cipher text secret using the plugin Vault `transit` wrapper.
```

File: examples/plugin-cli/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+done!
+```
+
+<hr>
+
+Build the example:
+```
+make 
+```
+Run the required [Vault's Transit Secrets
```
Comment (jefferai) [inline examples/plugin-cli/README.md:41]: I'm not 100% sure what you're meaning to say here in this paragraph but something is off :-)

**Review by jefferai — APPROVED** [review]
LGTM! Thank you!

File: examples/plugin-cli/README.md (hunk 1 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:4]: ty

File: examples/plugin-cli/README.md (hunk 2 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:6]: ty

File: examples/plugin-cli/README.md (hunk 3 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:7]: ty

File: examples/plugin-cli/README.md (hunk 4 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:18]: ty

File: examples/plugin-cli/README.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+via configuration but you didn't want to include every possible KMS wrapper
+dependency into your application.  Using go-plugin wrappers allows you to build
+a set of kms wrappers as plugins and perhaps embed the plugin binaries into your
+app (like we do in this example).  Then your app has no direct dependencies on
+the KMS's you wish to support.
+
+
+Running the cli will:
+- Initialize a go-plugin vault transit.
+- Encrypt a plaintext secret using the plugin vault transit wrapper.
```
[the remark this answers is not in the document]
Comment (jimlambrt) [reply examples/plugin-cli/README.md:19]: ty

File: examples/plugin-cli/README.md (hunk 5 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:20]: ty

File: examples/plugin-cli/README.md (hunk 6 above)
Comment (jimlambrt) [reply examples/plugin-cli/README.md:41]: chuckle.  ty.

**Pull request merged**